Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -328,44 +311,192 @@ extension TopTabsViewController: TabSelectionDelegate {
     func didSelectTabAtIndex(index: Int) {
         let tab = tabsToDisplay[index]
         tabManager.selectTab(tab)
-        collectionView.reloadData()
-        collectionView.setNeedsDisplay()
-        delegate?.topTabsDidChangeTab()
-        scrollToCurrentTab()
     }
 }
 
-extension TopTabsViewController : WKNavigationDelegate {
-    func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
-        collectionView.reloadData()
+// Collection Diff (animations)
+extension TopTabsViewController {
+
+    struct tableUpdate {
+        let reloads: Set<NSIndexPath>
+        let inserts: Set<NSIndexPath>
+        let deletes: Set<NSIndexPath>
+
+        init(updates: [tableUpdate]) {
```

Comment:
This seems odd using a list of `tableUpdates` to initialize a since `tableUpdate`. If you're looking to have a way of combining `tableUpdate`s together, how about overloading the + operator?